keras/engine/training.py

```diff
@@ -2023,6 +2023,8 @@ def generate_arrays_from_file(path):
                             ' and multiple workers may duplicate your data.'
                             ' Please consider using the`keras.utils.Sequence'
                             ' class.'))
+        if is_sequence:
```

Comment:
+@fchollet

This change messes up the logic of having less number of steps for special occasions, like [distributed training](https://github.com/uber/horovod/blob/master/examples/keras_mnist_advanced.py#L116) or some other cases.

Is it really necessary?  Can we use old logic of being able to explicitly pass the value that we want?

Reply:
It wasn't used. You would get into a weird behavior where the Enqueuer would go through all the dataset, but the fit_loop would only go through some. You can change the __len__ of your Sequence to do partial training.

Reply:
@Dref360, it's actually desired behavior to have Enqueuer keep enqueueing since next inner iteration inside the `fit_generator` would just keep reading the data - if I'm understanding the code correctly.

What's the user-friendly way of modifying `len` of `ImageDataGenerator` that would make it return random sub-permutations every epoch?